org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/navigation/XbaseHyperLinkHelper.java

```diff
@@ -171,8 +172,9 @@ protected void createHyperlinksInJavaDoc(XtextResource resource, int offset, IHy
 		EObjectInComment eObjectReferencedInComment = javaDocTypeReferenceProvider.computeEObjectReferencedInComment(resource, offset);
 		if(eObjectReferencedInComment != null) {
 			EObject target = eObjectReferencedInComment.getEObject();
-			ITextRegion region = eObjectReferencedInComment.getRegion();
-			createHyperlinksTo(resource, new Region(region.getOffset(), region.getLength()), target, acceptor);
+			ITextRegion textRegion = eObjectReferencedInComment.getRegion();
+			IRegion region = new Region(textRegion.getOffset(), textRegion.getLength());
+			createHyperlinksTo(resource, region, target, acceptor);
```

Comment:
this is what concerns me

Reply:
:(

Reply:
Actually the PR contains the following refactoring:
using the code
```
ITextRegion textRegion = eObjectReferencedInComment.getRegion();
IRegion region = new Region(textREgion.getOffset(), textRegion.getLength());
createHypelrinksTo(resource, region, target, acceptor);
```
instead of the code
`createHyperlinksTo(resource, new Region(region.getOffset(), region.getLength());`
the deprecation warning is gone.

@cdietrich Could you please elaborate a little bit more what exactly concerns you about this?

Reply:
if a user has customized createHyperlinksTo(Region) does the code still work?

Reply:
Ah, ok, I understand.

If the user has customized the createHyperlinksTo(Region) method and calls that like 
```
Region region = ...
createHyperlinksTo(region);
```
that method call will be bound to the customized createHyperlinksTo(Region) method.
```
But if the user calls the customized method like
IRegion region = ...
createHyperlinksTo(region);
```
that method call will be bound to the newly introduced (inherited) createHyperlinksTo(IRegion) method and the customized one will not be invoked.

So if we consider this, this change is indeed an API break. But the user will get a warning that a deprecated method is overridden/customized, so he(she (I hope) will understand what to do. 

@cdietrich  If you think that this PR causes more problem than it solves, then feel free to close it without merging it.

Reply:
@miklossy i wonder if you can apply a pattern like

https://github.com/eclipse/xtext-eclipse/blob/7b4d7e44c930eb8e9cee724f983d31cd36632b15/org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java#L549

to address the issue

Reply:
Thanks for the hint! Let me think about it....

Reply:
This also applies for the default helper that was not adapted